app/v2/models/responses/hateoas/HateoasMovementMessageIdsResponse.scala

```diff
@@ -32,20 +33,31 @@ object HateoasMovementMessageIdsResponse extends HateoasResponse {
         "self"                    -> Json.obj("href" -> getMessagesUri(movementId, receivedSince, movementType)),
         movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
       ),
-      "messages" -> messageIds.map(
+      "messages" -> messageIds.map {
         message =>
-          Json.obj(
-            "_links" -> Json.obj(
-              "self"                    -> Json.obj("href" -> getMessageUri(movementId, message.id, movementType)),
-              movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
-            ),
-            "id"                              -> message.id.value,
-            s"${movementType.movementType}Id" -> movementId.value,
-            "received"                        -> message.received,
-            "type"                            -> message.messageType,
-            "status"                          -> message.status
+          val jsMessageLinksObject = Json.obj(
+            "self"                    -> Json.obj("href" -> getMessageUri(movementId, message.id, movementType)),
+            movementType.movementType -> Json.obj("href" -> getMovementUri(movementId, movementType))
           )
-      )
+          if (message.status == MessageStatus.Pending) {
```

Comment:
I am not in the favour of hiding something that is available in database. Its look conflict with me. As if someone is checking with database they will find messageType but when they look in response they would not find it for large messageType. So as I commented on transit-movement backend api we should do this as part of callback. Its my suggestion. But if you guys think that is fine then no problem.

Reply:
The trader is _never_ going to be able to check the database. Remember the role of this service in this scenario is to create the view model, which does not, and frankly, should not, be driven by the database model.

I've not checked the code itself so I don't have an opinion of whether this is the "right" way to hide it, but it's perfectly fine for it to be done here.

Reply:
If the type is pending, we won't know the message type until we get the message - the status will change from Pending and therefore we will return the  message type